tidy3d/components/simulation.py

```diff
@@ -3777,23 +3777,26 @@ def _check_bloch_vec(
         if bloch_vec != expected_bloch_vec:
             test_val = np.real(expected_bloch_vec - bloch_vec)
 
-            if np.isclose(test_val % 1, 0) and not np.isclose(test_val, 0):
+            test_val_is_int = np.isclose(test_val, np.round(test_val))
+            src_name = f" '{source.name}'" if source.name else ""
+
+            if test_val_is_int and not np.isclose(test_val, 0):
                 # the given Bloch vector is offset by an integer
                 log.warning(
```

Comment:
can we just turn off this validator maybe? it is pretty annoying if you are setting things up properly